src/redis-cli.c

```diff
@@ -9332,9 +9487,13 @@ static void findHotKeys(void) {
         /* Now update our stats */
         for(i=0;i<keys->elements;i++) {
             sampled++;
-            /* Update overall progress */
-            if(sampled % 1000000 == 0) {
-                printf("[%05.2f%%] Sampled %llu keys so far\n", pct, sampled);
+
+            /* Only show the original progress output when writing to a file */
+            if (!(isatty(STDOUT_FILENO) || getenv("FAKETTY"))) {
```

Comment:
what about adding a macro like IS_TTY_OR_FAKETTY() to avoid repeated so many times.

Reply:
You are right, even if it will not reduce the number of lines, it will probably help the readability. Let me give it a try.

Reply:
Done. Actually, as the macro is shorter, it reduced some `if` statement from 2 lines to 1 line.